util/stmtsummary/statement_summary.go

```diff
@@ -879,13 +886,29 @@ func (ssElement *stmtSummaryByDigestElement) toDatum(ssbd *stmtSummaryByDigest)
 		break
 	}
 
+	var cols []types.Datum
 	// Actually, there's a small chance that endTime is out of date, but it's hard to keep it up to date all the time.
-	return types.MakeDatums(
+	timeAndType := types.MakeDatums(
 		types.NewTime(types.FromGoTime(time.Unix(ssElement.beginTime, 0)), mysql.TypeTimestamp, 0),
 		types.NewTime(types.FromGoTime(time.Unix(ssElement.endTime, 0)), mysql.TypeTimestamp, 0),
 		ssbd.stmtType,
-		ssbd.schemaName,
-		ssbd.digest,
+	)
+	cols = append(cols, timeAndType...)
+
+	var schemaNameAndDigest []types.Datum
+	if ssbd.isOtherSummary {
```

Comment:
Why not just initialize `digest` and `schemaName` as "" and then `convertEmptyToNil`?